FBAllocationTracker/Generations/FBAllocationTrackerGeneration.mm

```diff
@@ -45,12 +47,22 @@
         /**
          Retain object and add it to returnValue.
          This operation can be unsafe since we are retaining object that could
-         be deallocated on other thread.
+         be deallocated on other thread. 
+         
+         When NSZombie enabled, we can find if object has been deallocated by checking its class name.
          */
-        returnValue.push_back(object);
+        if (fb_isNSZombieEnabled()) {
+          if (!fb_isZombieObject(object)) {
```

Comment:
I wanted to make it clear that this zombie check makes sense only when NSZombieEnabled. For me it reads more clear. @Gricha do you want to simplify this `if` as @kastiglione suggests?